src/main/java/library/Member.java

```diff
@@ -53,75 +55,43 @@ public void getAllBooks() {
 
     @Override
     public List<BookItem> getBooksByAuthor(String author) {
-        List<BookItem> tempList = new ArrayList<>();
-        for (BookItem item : booksList) {
-            if (item.getAuthor().toString().contains(author)) {
-                tempList.add(item);
-                System.out.println(item.getTitle() + " : " + item.getAuthor());
-            }
-            else {
-                throw new AuthorNotFoundException("Author not found!");
-            }
-        }
         if (booksList.isEmpty()) {
             throw new BookNotFoundException("Book list is empty!");
         }
-        return tempList;
+        booksList.stream()
+                .filter(name -> name.getAuthor().toString().contains(author))
+                .forEach(bookItem -> System.out.format("%s : %s%n", bookItem.getTitle(), bookItem.getAuthor()));
+        return booksList;
```

Comment:
according to method name getBooksByAuthor you should return only books written by author you've sent in a parameter
as i can see, you print those books to console and then you return full book list

Reply:
fixed